jasmine/spec/inverted-index-test.js

```diff
@@ -2,59 +2,71 @@
-  it('reads a JSON file and asserts that it not empty', () => {
-    expect(invertedIndexTest.validFiles(emptyFile)[1]).toBe('Error, empty file');
-  });
-  it('reads a JSON file and asserts that it has the appropriate properties', () => {
-    expect(invertedIndexTest.validFiles(wrongFile)[1]).toBe('Invalid JSON file');
-  });
-  it('reads a JSON file and asserts that it has text and title properties', () => {
-    expect(invertedIndexTest.validFiles(wrongFile.title)[1]).toBe('Invalid JSON file');
+  it('reads a JSON file and asserts that its not empty',
+  () => {
+    expect(invertedIndexTest.validFiles(emptyFile).statusMessage)
+    .toBe('Error, empty file');
   });
-  it('parses received files into JSON objects', () => {
-    expect(typeof (invertedIndexTest.validFiles(validBook))).toBe(typeof {});
-  });
-  it('reads a JSON file and asserts that it valid', () => {
-    expect(invertedIndexTest.validFiles(JSON.stringify(validBook))[1]).toBe('Success');
-  });
-  it('reads a file and asserts that it invalid', () => {
-    expect(invertedIndexTest.validFiles(JSON.stringify(trollTest))[1]).toBe('this Index takes books with Title and Text property only');
+
+  it('reads a JSON file and asserts that it is valid',
+  () => {
+    expect(invertedIndexTest.validFiles(JSON.stringify(validBook))
+    .statusMessage).toBe('Success');
   });
-  it('reads a JSON file and asserts that it invalid', () => {
-    expect(invertedIndexTest.validFiles(JSON.stringify(wrongFile))[1]).toBe('this Index takes books with Title and Text property only');
+
+  it('reads a JSON file and asserts that it has the correct properties',
+  () => {
+    expect(invertedIndexTest.validFiles(JSON.stringify(wrongFile))
+    .statusMessage)
+    .toBe('this Index takes books with Title and Text property only');
   });
 });
 
 describe('Populate Index', () => {
-  it('verifies that an Index is created once the JSON file has been read', () => {
-    expect(invertedIndexTest.createIndex('booky.json', validBook)['booky.json'].alice).toBeDefined();
+  invertedIndexTest.createIndex('books.json', validBook);
+  it('verifies that an Index is created once the JSON file has been read',
+  () => {
+    expect(invertedIndexTest.getIndex()).toBeDefined();
   });
 
-  it('should map the string keys to the correct objects in the JSON array', () => {
-    expect(invertedIndexTest.createIndex('books.json', validBook)['books.json'].of).toEqual({ 0: true, 1: true });
+  it('should accept an optional parameter in the get index',
+  () => {
+    expect(invertedIndexTest.getIndex('books.json')).toBeDefined();
   });
-  it('Should confirm correct index is returned', () => {
-    expect(invertedIndexTest.getIndex()['booky.json'].of).toEqual({ 0: true, 1: true });
-    expect(Object.keys(invertedIndexTest.getIndex()['booky.json']).length).toEqual(25);
+
+  it('should map the string keys to the correct objects in the JSON array',
+  () => {
+    expect(invertedIndexTest.getIndex()['books.json']
```

Comment:
### Critical

You only corrected `.getIndex()['books.json']` in one instance, you should pass `books.json` as an argument!